test/cli/delegated_routing_http_test.go

```diff
@@ -0,0 +1,123 @@
+package cli
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/test/cli/harness"
+	. "github.com/ipfs/kubo/test/cli/testutils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHTTPDelegatedRouting(t *testing.T) {
+	t.Parallel()
+	h := harness.NewT(t)
+	node := h.Cluster.InitSingle()
+	node.Start()
+
+	fakeServer := func(resp string) *httptest.Server {
+		return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := w.Write([]byte(resp))
+			if err != nil {
+				panic(err)
+			}
+		}))
+	}
+
+	findProvsCID := "baeabep4vu3ceru7nerjjbk37sxb7wmftteve4hcosmyolsbsiubw2vr6pqzj6mw7kv6tbn6nqkkldnklbjgm5tzbi4hkpkled4xlcr7xz4bq"
+	prov := "12D3KooWARYacCc6eoCqvsS9RW9MA2vo51CV75deoiqssx3YgyYJ"
+
+	t.Run("default routing config has no routers defined", func(t *testing.T) {
+		assert.Nil(t, node.ReadConfig().Routing.Routers)
+	})
+
+	t.Run("no routers means findprovs returns no results", func(t *testing.T) {
+		res := node.IPFS("routing", "findprovs", findProvsCID).Stdout.String()
+		assert.Empty(t, res)
+	})
+
+	t.Run("no routers means findprovs returns no results", func(t *testing.T) {
+		res := node.IPFS("routing", "findprovs", findProvsCID).Stdout.String()
+		assert.Empty(t, res)
+	})
+
+	node.Stop()
+
+	t.Run("missing method params make the daemon fail", func(t *testing.T) {
+		node.UpdateConfig(func(cfg *config.Config) {
+			cfg.Routing.Type = config.NewOptionalString("custom")
+			cfg.Routing.Methods = config.Methods{
+				"find-peers":     {RouterName: "TestDelegatedRouter"},
+				"find-providers": {RouterName: "TestDelegatedRouter"},
+				"get-ipns":       {RouterName: "TestDelegatedRouter"},
+				"provide":        {RouterName: "TestDelegatedRouter"},
+			}
+		})
```

Comment:
Just a side note: this test demonstrates two different ways to get at the config. This way is reading the config file from disk and parsing it straight into a `config.Config` struct. The way below calls the CLI and returns the raw string. The `config.Config` struct is more useful for setting up config for a test, but if you want to test the ability set and read config itself, then use the CLI.